### PR TITLE
Added UptimeMonitor tile

### PR DESCRIPTION
**Pull request by AlexVanderbist**

### CODE REVIEW

File: resources/assets/js/components/UptimeMonitor.vue [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    export default {
+
+        components: {
+            Grid,
+        },
+
+        filters: {
+            formatDuration(start) {
+                return moment.duration(moment().diff(start), 'milliseconds').format("d[d] h[h] m[m]");
```
Comment (freekmurze) [inline resources/assets/js/components/UptimeMonitor.vue]: Add this functionality to a new function in `helpers.js` and add some tests to make sure it works properly.

File: resources/assets/js/components/UptimeMonitor.vue [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                        this.online(response.url);
+                    },
+                    'UptimeMonitor.UptimeCheckFailed': response => {
+                        this.offline(response.url, response.startedFailingAt);
+                    },
+                };
+            },
+
+            online(url) {
+                this.offlineUrls = this.offlineUrls.filter(offlineUrl => url != offlineUrl.url);
```
Comment (freekmurze) [inline resources/assets/js/components/UptimeMonitor.vue]: Rename `offlineUrls` to `failingUrls` so it matches a bit more with the event name.

File: resources/assets/js/components/UptimeMonitor.vue [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        },
+
+        methods: {
+            getEventHandlers() {
+                return {
+                    'UptimeMonitor.UptimeCheckSucceeded': response => {
+                        this.online(response.url);
+                    },
+                    'UptimeMonitor.UptimeCheckRecovered': response => {
+                        this.online(response.url);
```
Comment (freekmurze) [inline resources/assets/js/components/UptimeMonitor.vue]: rename `online` to `show`

File: resources/assets/js/components/UptimeMonitor.vue [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                    },
+                    'UptimeMonitor.UptimeCheckFailed': response => {
+                        this.offline(response.url, response.startedFailingAt);
```
Comment (freekmurze) [inline resources/assets/js/components/UptimeMonitor.vue]: rename `offline` to `remove`

File: resources/views/dashboard.blade.php [hunk 5]
```diff
@@ -12,7 +12,8 @@
         <github-file file-name="seb" grid="d1"></github-file>
         <github-file file-name="rogier" grid="c2"></github-file>
         <github-file file-name="willem" grid="d2"></github-file>
-        <internet-connection grid="c3"></internet-connection>
+        {{--<internet-connection grid="c3"></internet-connection>--}}
```
Comment (freekmurze) [inline resources/views/dashboard.blade.php]: I don't like dead code, just remove it 😄

File: resources/assets/js/components/UptimeMonitor.vue (hunk 3 above)
Comment (AlexVanderbist) [inline resources/assets/js/components/UptimeMonitor.vue]: Shouldn't it be `remove` since the site came back online? Same for the `offline` method

**Comment by sebastiandedeyne** [comment]
👏

**Comment by freekmurze** [comment]
👍

**Comment by m1guelpf** [comment]
How do I set up this?

**Comment by freekmurze** [comment]
Backend: We've used our own [uptime monitor package](https://docs.spatie.be/laravel-uptime-monitor/v2/introduction) for this. It's installed in a separate application (in a private repo, it's not included in this dashboard repo). That application configured it in such way that all it's events are broadcasted to the same pusher channel that the dashboard is listening to.

Eg. when the uptime monitor detects that a site is down it's [UptimeCheckFailed](https://docs.spatie.be/laravel-uptime-monitor/v2/monitoring-uptime/events#uptimecheckfailed) is being broadcasted.

Frontend: it's just a Vue component that listens to incoming `UptimeCheckFailed` events.

**Comment by SimplyCorey** [comment]
Can I see the example code to fire the broadcast please?

**Pull request merged**